arrow/table_buffer.go

```diff
@@ -26,25 +26,46 @@ type TableBuffer struct {
 var _ flux.ColReader = (*TableBuffer)(nil)
 
 func (t *TableBuffer) Len() int {
+	if len(t.Values) == 0 {
+		return 0
+	}
 	return t.Values[0].Len()
 }
 
 func (t *TableBuffer) Bools(j int) *array.Boolean {
+	if len(t.Values) == 0 {
+		return nil
```

Comment:
Researching this now, but I'm not sure this is valid to do.

Reply:
Don't see why not. Seems legit to me.

Reply:
The issue I found is that it appears some parts of the code then refer to this nil. I thought they would check the length or if the table was empty, but they grab the individual column and check the length of that which causes a panic.